src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutMissingTagCheck.java

```diff
@@ -0,0 +1,193 @@
+                MAX_THRESHOLD_DEGREES_DEFAULT, Angle::degrees);
+        this.minAngleThreshold = configurationValue(configuration, "threshold.degrees",
+                MIN_THRESHOLD_DEGREES_DEFAULT, Angle::degrees);
+    }
+
+    /**
+     * Validates if given {@link AtlasObject} is actually an {@link Edge} and is a potential
+     * roundabout and also corresponding OSM identifier shouldn't be flagged before (this is for
+     * avoiding duplicate flags)
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge && !this.isFlagged(object.getOsmIdentifier())
+                && object.getIdentifier() % MODULUS == FIRST_EDGE_SECTION
+                && ((Edge) object).isMasterEdge() && HighwayTag.isCarNavigableHighway(object)
+                && object.getTag(JunctionTag.KEY).isEmpty() && object.getTag(AreaTag.KEY).isEmpty()
+                && isPartOfClosedWay((Edge) object) && intersectingWithMoreThan((Edge) object);
+    }
+
+    /**
+     * Flags an {@link Edge} is its circular shape and connected to at least
+     * {@link RoundaboutMissingTagCheck#MINIMUM_INTERSECTION} navigable roads. {@link Edge} doesn't
+     * have junction=roundabout or area=yes tags.
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Edge edge = (Edge) object;
+
+        final PolyLine originalGeom = buildOriginalOsmWayGeometry(edge);
+        // check maximum angle
+        final List<Tuple<Angle, Location>> maxOffendingAngles = originalGeom
+                .anglesGreaterThanOrEqualTo(this.maxAngleThreshold);
+        // check minimum angle
+        final List<Tuple<Angle, Location>> minOffendingAngles = originalGeom
+                .anglesLessThanOrEqualTo(this.minAngleThreshold);
+
+        if (maxOffendingAngles.isEmpty() && minOffendingAngles.isEmpty())
+        {
+            this.markAsFlagged(object.getOsmIdentifier());
+            return Optional.of(createFlag(new OsmWayWalker(edge).collectEdges(),
+                    this.getLocalizedInstruction(0)));
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Build original OSW way geometry from all MasterEdge sections
+     *
+     * @param edge
+     *            entity to check
+     * @return original Way geometry polyline
+     */
+    private PolyLine buildOriginalOsmWayGeometry(final Edge edge)
+    {
+        // Identify all sections of original OSM way
+        final Set<Edge> edgesFormingOSMWay = new OsmWayWalker(edge).collectEdges().stream()
+                .filter(Edge::isMasterEdge).collect(Collectors.toSet());
+
+        // Sort sectioned Edges by Ids
+        final List<Edge> sortedEdges = new ArrayList<>(edgesFormingOSMWay);
+        sortedEdges.sort(Comparator.comparingLong(AtlasObject::getIdentifier));
```

Comment:
nit: these could be combined to one statement  `new OsmWayWalker(edge).collectEdges().stream()
                .filter(Edge::isMasterEdge).distinct()
                .sorted(Comparator.comparingLong(AtlasObject::getIdentifier))
                .collect(Collectors.toList());`